src/main.cpp

```diff
@@ -2822,6 +3021,16 @@ bool CBlock::AcceptBlock(CValidationState &state, CDiskBlockPos *dbp)
                 return state.Invalid(error("AcceptBlock() : rejected nVersion=2 block"));
             }
         }
+        // Reject non-signaling blocks when 95% (75% on testnet) of
+        // the network has upgraded (until timeout activation):
+        if ((nHeight < APU_ACTIVATION_HEIGHT) && ((nVersion >> 28) != 3))
+        {
+            if ((!fTestNet && CBlockIndex::IsSuperMajorityBit(28, pindexPrev, 950, 1000)) ||
+                (fTestNet && CBlockIndex::IsSuperMajorityBit(28, pindexPrev, 75, 100)))
+            {
+                return state.Invalid(error("AcceptBlock() : rejected non-APU block before activation timeout"));
```

Comment:
I don't think this is necessary or compatible with bip9/bip8. After APU_ACTIVATION_HEIGHT, it will be activated regardless of the signaling. BIP8 doesn't force signaling like bip148.

Reply:
Yes, that's the point. Much like in BIP8, which activates on timeout as well.

Note the goal here is compatibility with version bits as used in common mining software, not matching the BIP8 or 9 state machine protocol exactly. The context is that most of the freicoin blocks being mined right now have version bits headers, so the old-style IsSuperMajority fork method won't work as it would prematurely activate.